presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1133,21 +1078,25 @@ private static ExpressionAnalysis analyzeExpressions(
             Metadata metadata,
             SqlParser sqlParser,
             RelationType tupleDescriptor,
-            Iterable<? extends Expression> expressions)
+            Iterable<Expression> expressions)
     {
         // expressions at this point can not have sub queries so deny all access checks
         // in the future, we will need a full access controller here to verify access to functions
-        ExpressionAnalyzer analyzer = create(new Analysis(), session, metadata, sqlParser, new DenyAllAccessControl(), false);
+        Node root = Iterables.size(expressions) > 0 ? ExpressionUtils.and(expressions) : TRUE_LITERAL;
```

Comment:
When somebody calls `Analysis.getScope(node)` and such `node` has not have mapping to scope. Then to find its scope a visitor is used, it finds the nearest parent node of this `node` which has a scope set and returns it as a `node` scope. This search is started from the root node.

Here I wanted to create such `artificial` root for all the expressions. But now, when you have asked I realized that is not needed and what is worse it introduces unnecessary confusion. So I removed that. Instead I create multiple roots, one per each expression.